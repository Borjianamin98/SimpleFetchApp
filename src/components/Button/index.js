import PropTypes from "prop-types";
import React from "react";
import {withLoading} from "../helper/withLoading";

export default class Button extends React.Component {
    render() {
        let {onClick, className, children} = this.props;
        return <button
            onClick={onClick}
            className={className}
            type="button"
        >
            {children}
        </button>;
    }
}

Button.defaultProps = {
    className: '',
};

Button.propTypes = {
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
    children: PropTypes.node.isRequired,
};

export const ButtonWithLoading = withLoading(Button);