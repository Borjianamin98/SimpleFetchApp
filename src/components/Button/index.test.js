import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16.1";
import Button from "./index.js";

Enzyme.configure({adapter: new Adapter()});

describe('Button', () => {
    const customButton = <Button
        onClick={(event) => console.log(event)}
        onDismiss={(event) => console.log(event)}>
        Give Me More
    </Button>

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(customButton, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('has a valid snapshot', () => {
        const component = renderer.create(customButton);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});