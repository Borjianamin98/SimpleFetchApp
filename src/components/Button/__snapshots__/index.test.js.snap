// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  Give Me More
</button>
`;
